cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/WildcardAuthorizable.java

```diff
@@ -0,0 +1,75 @@
+package co.cask.cdap.security.authorization.sentry.binding;
+
+import co.cask.cdap.security.authorization.sentry.model.Authorizable;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * This class supports wildcard matching of {@link Authorizable}s.
+ * "*" and "?" are the supported wildcards, "*" matches any number of characters and "?" matches one character.
+ */
+class WildcardAuthorizable {
```

Comment:
The existing `WildcardPrivilege` class uses different interface since it runs in Sentry. This class has to run in the extension, hence won't be able to modify the existing class. Also I did not add this class to the model package since it cannot be run in Sentry.